core/src/main/java/hudson/model/Queue.java

```diff
@@ -1414,7 +1423,7 @@ public void maintain() {
                             p.task.getFullDisplayName());
                     p.isPending = false;
                     pendings.remove(p);
-                    makeBuildable(p);
+                    makeBuildable(p); // TODO whatever this is for, the return value is being ignored, so this does nothing at all
```

Comment:
Looks like an unrelated bug. See #2510; we occasionally get here, but I have never figured out what this case means, and apparently this line does absolutely nothing. Maybe a regression from #1513? Clearly not covered by any existing tests.